INSTALL.rst

```diff
@@ -0,0 +1,437 @@
+``./configure.sh --static --no-static-binary``.
+
+
+Cross-compiling for Windows
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Cross-compiling cvc5 with Mingw-w64 can be done as follows:
+
+.. code:: bash
+
+  ./configure.sh --win64 --static <configure options...>
+
+  cd <build_dir>   # default is ./build
+  make             # use -jN for parallel build with N threads
+
+The built binary ``cvc5.exe`` is located in ``<build_dir>/bin`` and the cvc5
+library can be found in ``<build_dir>/lib``.
+
+
+Build dependencies
+------------------
+
+cvc5 makes uses of a number of tools and libraries. Some of these are required
+while others are only used with certain configuration options. If
+``--auto-download`` is given, cvc5 can automatically download and build most
+libraries that are not already installed on your system. If your libraries are
+installed in a non-standard location, you can use ``--dep-path`` to define an
+additional search path for all dependencies. Versions given are minimum
+versions; more recent versions should be compatible.
+
+- `GNU C and C++ (gcc and g++) <https://gcc.gnu.org>`_
+  or `Clang <https://clang.llvm.org>`_ (reasonably recent versions)
+- `CMake >= 3.9 <https://cmake.org>`_
+- `Python 3.x <https://www.python.org>`_
+  + module `toml <https://pypi.org/project/toml/>`_
+- `GMP v6.1 (GNU Multi-Precision arithmetic library) <https://gmplib.org>`_
+- `ANTLR 3.4 <http://www.antlr3.org/>`_
+- `Java >= 1.6 <https://www.java.com>`_
+- `SymFPU <https://github.com/martin-cs/symfpu/tree/CVC4>`_
+
+
+ANTLR 3.4 parser generator
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For most systems, the package manager no longer contains pre-packaged versions
+of ANTLR 3.4. With ``--auto-download``, cvc5 will automatically download and
+build ANTLR 3.4.
+
+
+GMP (GNU Multi-Precision arithmetic library)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+GMP is usually available on your distribution and should be used from there. If
+it does not, or you want to cross-compile, or you want to build cvc5 statically
+but the distribution does not ship static libraries, cvc5 builds GMP
+automatically when ``--auto-download`` is given.
+
+
+SymFPU (Support for the Theory of Floating Point Numbers)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`SymFPU <https://github.com/martin-cs/symfpu/tree/CVC4>`_ is an implementation
+of SMT-LIB/IEEE-754 floating-point operations in terms of bit-vector operations.
+It is required for supporting the theory of floating-point numbers and can be
+downloaded and built automatically.
+
+
+Optional Dependencies
+---------------------
+
+
+CaDiCaL (Optional SAT solver)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`CaDiCaL <https://github.com/arminbiere/cadical>`_ is a SAT solver that can be
+used for solving non-incremental bit-vector problems with eager bit-blasting.
+This dependency may improve performance. It can be downloaded and built
+automatically. Configure cvc5 with ``configure.sh --cadical`` to build with this
+dependency.
```

Comment:
```suggestion
`CaDiCaL <https://github.com/arminbiere/cadical>`_ is a SAT solver that can be
is used for the bit-vector solver. It can be downloaded and built
automatically.
```

Reply:
Also, it should be moved up to the required dependencies.